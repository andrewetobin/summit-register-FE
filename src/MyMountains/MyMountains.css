.MyMountains {
  background: url("../utilities/Images/mountains-2.jpg") no-repeat center
  rgba(165, 165, 165, 0.68);
  background-size: auto 100%;
  overflow: scroll;
}


.mountain-container {
  margin: 0 5%;
  /* width: 90%; */
}

.my-mountain {
  background: grey;
  border: 2px solid white;
  border-radius: 20px;
  color: white;
  margin: 10px auto;
  max-width: 60vw;
  opacity: .9;
  padding: 10px;
  text-align: center;
}

