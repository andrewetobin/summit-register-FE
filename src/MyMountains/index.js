import React, { Component } from "react";
import MyMountainRegister from "../MyMountainRegister";
import { getMyMountains } from "../utilities/helper/apiCalls";
import PropTypes from "prop-types";
import "./MyMountains.css";

class MyMountains extends Component {
  constructor() {
    super();
    this.state = { cleanMountainData: {} };
  }

  componentDidMount = async () => {
    const userId = this.props.userCredentials.data.id;
    const myMountains = await getMyMountains(userId);
    this.cleanUserMountainData(myMountains);
  };

  cleanUserMountainData = myMountains => {
    const cleanMountainData = myMountains.reduce((acc, log) => {
      if (!acc[log.attributes.mountain]) {
        acc[log.attributes.mountain] = [];
      }
      acc[log.attributes.mountain].push(log);
      return acc;
    }, {});
    this.setState({ cleanMountainData });
  };

  render() {
    const myMountainNames = Object.keys(this.state.cleanMountainData);

    const mountainCard = myMountainNames.sort().map(mountain => {
      return (
        <div className="my-mountain" key={mountain}>
          <h3>{mountain}</h3>
          {this.state.cleanMountainData[mountain].map(log => {
            return (
              <MyMountainRegister
                key={log.id}
                log={log.attributes}
              />
            );
          })}
        </div>
      );
    });

    return <div className="MyMountains">{mountainCard}</div>;
  }
}

MyMountains.propTypes = {
  validateSignIn: PropTypes.func,
  userCredentials: PropTypes.object
};

export default MyMountains;
