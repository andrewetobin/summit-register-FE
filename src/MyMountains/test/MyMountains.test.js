import React from "react";
import ReactDOM from "react-dom";
import MyMountains from "..";
import { shallow } from "enzyme";

describe("MyMountains", () => {
  let wrapper = shallow(<MyMountains validateSignIn={jest.fn()} />);
  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<MyMountains />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  describe("cleanUserMountainData", () => {
    const registries = [
      {
        attributes: {
          id: "9",
          name: "Chris",
          hometown: "Denver",
          comments: "Hi",
          image_url:
            "https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg",
          sign_time: "1/1/2019 @ 17:47:41",
          mountain: "mt jimbob"
        }
      },
      {
        attributes: {
          id: "9",
          name: "Chris",
          hometown: "Denver",
          comments: "Hi again",
          image_url:
            "https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg",
          sign_time: "1/2/2019 @ 17:47:41",
          mountain: "mt jimbob"
        }
      },
      {
        attributes: {
          id: "9",
          name: "Chris",
          hometown: "Denver",
          comments: "Hi",
          image_url:
            "https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg",
          sign_time: "1/2/2019 @ 17:47:41",
          mountain: "mt timbucktoo"
        }
      }
    ];
    it("should return a single object with mountainNames as Keys and registries as values", () => {
      wrapper.instance().cleanUserMountainData(registries);
      const { cleanMountainData } = wrapper.state();
      expect(cleanMountainData.hasOwnProperty("mt timbucktoo")).toBe(true);
      expect(cleanMountainData.hasOwnProperty("mt jimbob")).toBe(true);
      expect(cleanMountainData["mt jimbob"]).toHaveLength(2);
      expect(cleanMountainData["mt timbucktoo"]).toHaveLength(1);
    });
    it("should call cleanUserMountainData on ComponentDidMount", () => {
      expect(wrapper.instance().cleanUserMountainData).toBeCalled();
    });
  });
});
