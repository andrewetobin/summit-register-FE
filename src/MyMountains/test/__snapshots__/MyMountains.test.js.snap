// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyMountains should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyMountains
    validateSignIn={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="mountain-container"
      >
        <div
          className="comment"
        >
          <h4>
            Name: 
            Chris
          </h4>
          <h4>
            Hometown: 
            Denver
          </h4>
          <h4>
            Comments: 
            Hi
             
          </h4>
          <h4>
            Date: 
            1/1/2019 @ 17:47:41
          </h4>
          <img
            alt="user image"
            className="user-image"
            src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
          />
        </div>
        <div
          className="comment"
        >
          <h4>
            Name: 
            Chris
          </h4>
          <h4>
            Hometown: 
            Denver
          </h4>
          <h4>
            Comments: 
            Hi again
             
          </h4>
          <h4>
            Date: 
            1/2/2019 @ 17:47:41
          </h4>
          <img
            alt="user image"
            className="user-image"
            src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
          />
        </div>
        <div
          className="comment"
        >
          <h4>
            Name: 
            Chris
          </h4>
          <h4>
            Hometown: 
            Denver
          </h4>
          <h4>
            Comments: 
            Hi
             
          </h4>
          <h4>
            Date: 
            1/2/2019 @ 17:47:41
          </h4>
          <img
            alt="user image"
            className="user-image"
            src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
          />
        </div>
      </section>,
      "className": "MyMountains",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="comment"
          >
            <h4>
              Name: 
              Chris
            </h4>
            <h4>
              Hometown: 
              Denver
            </h4>
            <h4>
              Comments: 
              Hi
               
            </h4>
            <h4>
              Date: 
              1/1/2019 @ 17:47:41
            </h4>
            <img
              alt="user image"
              className="user-image"
              src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
            />
          </div>,
          <div
            className="comment"
          >
            <h4>
              Name: 
              Chris
            </h4>
            <h4>
              Hometown: 
              Denver
            </h4>
            <h4>
              Comments: 
              Hi again
               
            </h4>
            <h4>
              Date: 
              1/2/2019 @ 17:47:41
            </h4>
            <img
              alt="user image"
              className="user-image"
              src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
            />
          </div>,
          <div
            className="comment"
          >
            <h4>
              Name: 
              Chris
            </h4>
            <h4>
              Hometown: 
              Denver
            </h4>
            <h4>
              Comments: 
              Hi
               
            </h4>
            <h4>
              Date: 
              1/2/2019 @ 17:47:41
            </h4>
            <img
              alt="user image"
              className="user-image"
              src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
            />
          </div>,
        ],
        "className": "mountain-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Name: 
                Chris
              </h4>,
              <h4>
                Hometown: 
                Denver
              </h4>,
              <h4>
                Comments: 
                Hi
                 
              </h4>,
              <h4>
                Date: 
                1/1/2019 @ 17:47:41
              </h4>,
              <img
                alt="user image"
                className="user-image"
                src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
              />,
            ],
            "className": "comment",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Name: ",
                  "Chris",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Name: ",
                "Chris",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Hometown: ",
                  "Denver",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Hometown: ",
                "Denver",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Comments: ",
                  "Hi",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Comments: ",
                "Hi",
                " ",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Date: ",
                  "1/1/2019 @ 17:47:41",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Date: ",
                "1/1/2019 @ 17:47:41",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "user image",
                "className": "user-image",
                "src": "https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Name: 
                Chris
              </h4>,
              <h4>
                Hometown: 
                Denver
              </h4>,
              <h4>
                Comments: 
                Hi again
                 
              </h4>,
              <h4>
                Date: 
                1/2/2019 @ 17:47:41
              </h4>,
              <img
                alt="user image"
                className="user-image"
                src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
              />,
            ],
            "className": "comment",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Name: ",
                  "Chris",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Name: ",
                "Chris",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Hometown: ",
                  "Denver",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Hometown: ",
                "Denver",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Comments: ",
                  "Hi again",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Comments: ",
                "Hi again",
                " ",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Date: ",
                  "1/2/2019 @ 17:47:41",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Date: ",
                "1/2/2019 @ 17:47:41",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "user image",
                "className": "user-image",
                "src": "https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Name: 
                Chris
              </h4>,
              <h4>
                Hometown: 
                Denver
              </h4>,
              <h4>
                Comments: 
                Hi
                 
              </h4>,
              <h4>
                Date: 
                1/2/2019 @ 17:47:41
              </h4>,
              <img
                alt="user image"
                className="user-image"
                src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
              />,
            ],
            "className": "comment",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Name: ",
                  "Chris",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Name: ",
                "Chris",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Hometown: ",
                  "Denver",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Hometown: ",
                "Denver",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Comments: ",
                  "Hi",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Comments: ",
                "Hi",
                " ",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Date: ",
                  "1/2/2019 @ 17:47:41",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Date: ",
                "1/2/2019 @ 17:47:41",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "user image",
                "className": "user-image",
                "src": "https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="mountain-container"
        >
          <div
            className="comment"
          >
            <h4>
              Name: 
              Chris
            </h4>
            <h4>
              Hometown: 
              Denver
            </h4>
            <h4>
              Comments: 
              Hi
               
            </h4>
            <h4>
              Date: 
              1/1/2019 @ 17:47:41
            </h4>
            <img
              alt="user image"
              className="user-image"
              src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
            />
          </div>
          <div
            className="comment"
          >
            <h4>
              Name: 
              Chris
            </h4>
            <h4>
              Hometown: 
              Denver
            </h4>
            <h4>
              Comments: 
              Hi again
               
            </h4>
            <h4>
              Date: 
              1/2/2019 @ 17:47:41
            </h4>
            <img
              alt="user image"
              className="user-image"
              src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
            />
          </div>
          <div
            className="comment"
          >
            <h4>
              Name: 
              Chris
            </h4>
            <h4>
              Hometown: 
              Denver
            </h4>
            <h4>
              Comments: 
              Hi
               
            </h4>
            <h4>
              Date: 
              1/2/2019 @ 17:47:41
            </h4>
            <img
              alt="user image"
              className="user-image"
              src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
            />
          </div>
        </section>,
        "className": "MyMountains",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="comment"
            >
              <h4>
                Name: 
                Chris
              </h4>
              <h4>
                Hometown: 
                Denver
              </h4>
              <h4>
                Comments: 
                Hi
                 
              </h4>
              <h4>
                Date: 
                1/1/2019 @ 17:47:41
              </h4>
              <img
                alt="user image"
                className="user-image"
                src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
              />
            </div>,
            <div
              className="comment"
            >
              <h4>
                Name: 
                Chris
              </h4>
              <h4>
                Hometown: 
                Denver
              </h4>
              <h4>
                Comments: 
                Hi again
                 
              </h4>
              <h4>
                Date: 
                1/2/2019 @ 17:47:41
              </h4>
              <img
                alt="user image"
                className="user-image"
                src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
              />
            </div>,
            <div
              className="comment"
            >
              <h4>
                Name: 
                Chris
              </h4>
              <h4>
                Hometown: 
                Denver
              </h4>
              <h4>
                Comments: 
                Hi
                 
              </h4>
              <h4>
                Date: 
                1/2/2019 @ 17:47:41
              </h4>
              <img
                alt="user image"
                className="user-image"
                src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
              />
            </div>,
          ],
          "className": "mountain-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Name: 
                  Chris
                </h4>,
                <h4>
                  Hometown: 
                  Denver
                </h4>,
                <h4>
                  Comments: 
                  Hi
                   
                </h4>,
                <h4>
                  Date: 
                  1/1/2019 @ 17:47:41
                </h4>,
                <img
                  alt="user image"
                  className="user-image"
                  src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
                />,
              ],
              "className": "comment",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Name: ",
                    "Chris",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Name: ",
                  "Chris",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Hometown: ",
                    "Denver",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Hometown: ",
                  "Denver",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Comments: ",
                    "Hi",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Comments: ",
                  "Hi",
                  " ",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Date: ",
                    "1/1/2019 @ 17:47:41",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Date: ",
                  "1/1/2019 @ 17:47:41",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "user image",
                  "className": "user-image",
                  "src": "https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Name: 
                  Chris
                </h4>,
                <h4>
                  Hometown: 
                  Denver
                </h4>,
                <h4>
                  Comments: 
                  Hi again
                   
                </h4>,
                <h4>
                  Date: 
                  1/2/2019 @ 17:47:41
                </h4>,
                <img
                  alt="user image"
                  className="user-image"
                  src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
                />,
              ],
              "className": "comment",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Name: ",
                    "Chris",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Name: ",
                  "Chris",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Hometown: ",
                    "Denver",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Hometown: ",
                  "Denver",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Comments: ",
                    "Hi again",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Comments: ",
                  "Hi again",
                  " ",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Date: ",
                    "1/2/2019 @ 17:47:41",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Date: ",
                  "1/2/2019 @ 17:47:41",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "user image",
                  "className": "user-image",
                  "src": "https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Name: 
                  Chris
                </h4>,
                <h4>
                  Hometown: 
                  Denver
                </h4>,
                <h4>
                  Comments: 
                  Hi
                   
                </h4>,
                <h4>
                  Date: 
                  1/2/2019 @ 17:47:41
                </h4>,
                <img
                  alt="user image"
                  className="user-image"
                  src="https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg"
                />,
              ],
              "className": "comment",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Name: ",
                    "Chris",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Name: ",
                  "Chris",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Hometown: ",
                    "Denver",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Hometown: ",
                  "Denver",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Comments: ",
                    "Hi",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Comments: ",
                  "Hi",
                  " ",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Date: ",
                    "1/2/2019 @ 17:47:41",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Date: ",
                  "1/2/2019 @ 17:47:41",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "user image",
                  "className": "user-image",
                  "src": "https://res.cloudinary.com/summit-register/image/upload/v1546476463/pe0msmw1hh91l5dv9kha.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
